packages/slate/test/interfaces/Editor/unhangRange/text-hanging.tsx

```diff
@@ -0,0 +1,28 @@
+/** @jsx jsx */
+import { Editor } from 'slate'
+import { jsx } from '../../..'
+
+export const input = (
+  <editor>
+    <block>
+      <text>
+        before
+        <anchor />
+      </text>
+      <text>selected</text>
+      <text>
+        <focus />
+        after
+      </text>
+    </block>
+  </editor>
+)
+
+export const test = editor => {
+  return Editor.unhangRange(editor, editor.selection)
+}
+
+export const output = {
+  anchor: { path: [0, 0], offset: 6 },
+  focus: { path: [0, 2], offset: 0 },
+}
```

Comment:
It doesn't perform unhanging to for text nodes.

Which actually prevents using it to work around inconsistent browsers behavior related to `onDOMSelectionChange`. 

Related tickets: #3336 #3374 #2317 #3840 #4634 

Would you be okay if extend `Editor.unhangRange()` to also unhang Text nodes?